### ISSUE TITLE
`--migrations` to specify output directory for migrations

### ISSUE THREAD

**Issue by janpio** [prisma/prisma#4720]
With `--schema` we can now specify the file path of where the Prisma Schema is located, instead of the default `./prisma/schema.prisma`. This has the unfortunate side effect, that migrations are also output next to that file - which might not be what you want.

We want to have a `--migrations` parameter that takes the path to a _folder_ where all the migration related content is output. The default path for that parameter, if it is not supplied, thus is `./prisma/migrations`.

**Comment by benhickson** [comment]
Any updates on this? commenting to subscribe :)

**Comment by Jolg42** [comment]
Hi there! @benhickson & others

I'm curious about "why" this feature is needed for you, so we can prioritize better.

Could you give example(s) where customizing the migrations folder (default is in ./prisma/migrations) is needed / better maybe?

**Comment by benhickson** [comment]
Hi @Jolg42 ! 

In our situation, we have a package in our monorepo for prisma. We have two databases, and each need their own prisma schema. Ideally we'd be able to commit migrations for both of them. As of now, we have to manually rename the migrations folder after running the migration, then commit that renamed folder, renaming it back to "migrations" when we want to run additional migrations. This would be unwieldy to do if we ran the migrations sequentially, but ideally we'd want to be able to run them concurrently, which wouldn't really work.

The prisma migrate command has a --schema flag that we're already using to point to our different schemas, ideally there'd be a corresponding flag to set the migrations folder.

As an alternative to the CLI flag, a block in the schema file would work, similar to how the `generator client { }` block has a `output` option. At least that would work for us, as our schemas and migrations are 1-to-1. Others may have setups that aren't 1-to-1.

Open to any and all feedback on how we could do things differently!

**Comment by Jolg42** [comment]
@benhickson Thank you for the detailed explanation!

I guess a workaround could be something like this, one folder per schema:
- ./src/db1/schema.prisma 
- ./src/db2/schema.prisma

And then calling prisma commands with `--schema="./src/db1/schema.prisma"`

Migrate will then create a migrations directory next to each schema like
- ./src/db1/migrations/...
- ./src/db2/migrations/...

Is that a good enough workaround or customizing the name of the migrations directory would be better here?

**Comment by benhickson** [comment]
@Jolg42 ah, that makes sense, seems so obvious!

so the migrations directory is always a sibling of the schema file? I think that would all work fine. I think the option would still be appreciated, but that should work for our use case.

**Comment by Jolg42** [comment]
@benhickson Yes how it's implemented is that the migrations directory path is "next" / "a sibling" to the schema.

Happy it works for you!

**Comment by dborstelmann** [comment]
I think this works now?  I have repo_root/db/schema.prisma and repo_root/db/migrations and it seems to work?

**Comment by Jolg42** [comment]
@dborstelmann I think what you are describing here is the behavior of `--schema="./repo_root/db/schema.prisma"`
See https://github.com/prisma/prisma/issues/4720#issuecomment-832819929

This indeed works. Though it's not possible to customize the name and location of the `migrations` directory independently.

**Comment by luwol03** [comment]
This could be archieved by adding a cli parameter and then set the output migrations path to that value like it is described here: https://github.com/prisma/prisma/discussions/12568#discussioncomment-2505709

**Comment by carlosLMZarego** [comment]
> Hi there! @benhickson & others
> 
> I'm curious about "why" this feature is needed for you, so we can prioritize better.
> 
> Could you give example(s) where customizing the migrations folder (default is in ./prisma/migrations) is needed / better maybe?

A good example is when you run the migration in AWS lambda and you want to use the /temp folder for instance

**Comment by w1am** [comment]
> Hi there

Sometimes, I write different migration files depending on my current environment. For example, I would like to populate my local database with mock data, which I wouldn't want to do in other environments. For now, I have written a script to duplicate the schema file into different folders named after my environment and specify the `--schema` path in the CLI.

**Comment by sacru2red** [comment]
I'm not sure if this is a suitable example,
MySQL does not support multiSchema. Therefore, users create multiple schema.prisma files and also store the output in multiple directories (#16943). As a result, there is a need to manage migration files in multiple parts (since each database is separate in MySQL).

**Comment by pandeymangg** [comment]
any progress on this?

**Comment by StiliyanKushev** [comment]
+1 Bump

Would love to see this feature come to life.

**Comment by batchor** [comment]
@jkomyno Really hope this feature can land soon!

**Comment by vitran96** [comment]
I am also look forward for this feature as we have a case to support both MySql and PostgreSql with same schema. Because depending on the client, we will have to switch DB.

**Comment by samislam** [comment]
It's been 5 years.

**Comment by Frost-on-Web** [comment]
Actually I can't understand why this is not a basic feature.

I agree it is mostly for convenience and personal folder organization but what is not ?

My "wanted" Prisma folder :

```
/prisma
   /schema
      /base.schema
      /user.schema
      /booking.schema
   /migrations
      /000000000000_squashed_migrations
      /202506232201_just_another_migration
   /seeds
      /main.ts
      /users.ts
```

My "wanted" package.json :

```json
"prisma": {
    "schema": "prisma/schema",
    "migrations": "prisma/migrations",
    "seed": "tsx prisma/seeds/main.ts"
},
```

Easy, clear, but I can't do it like this and I am a little disappointed.
And if there is one thing I don't like I do not like it is to be disappointed.

(Just joking, five points for Gryffondor if you have the reference)
Have a nice day !

**Comment by FGoessler** [comment]
Am happy announce that with Prisma 6.12.0 you can now specify your migrations folder location via `prisma.config.ts`. [See the documentation here](https://www.prisma.io/docs/orm/reference/prisma-config-reference#migrationspath).

We do not plan to add a CLI argument for this.

Am closing this issue now as the underlying problem should be solved with this. :)
